edb/pgsql/parser/ast_builder.py

```diff
@@ -369,27 +377,37 @@ def _build_variable_set_stmt(n: Node, c: Context) -> pgast.Statement:
         if name == "TRANSACTION" or name == "SESSION CHARACTERISTICS":
             return pgast.SetTransactionStmt(
                 options=_build_transaction_options(n["args"], c),
-                scope=pgast.OptionsScope.TRANSACTION
-                if name == "TRANSACTION"
-                else pgast.OptionsScope.SESSION,
+                scope=(
+                    pgast.OptionsScope.TRANSACTION
+                    if name == "TRANSACTION"
+                    else pgast.OptionsScope.SESSION
+                ),
             )
 
     if n["kind"] == "VAR_SET_VALUE":
         return pgast.VariableSetStmt(
             name=n["name"],
             args=pgast.ArgsList(args=_list(n, c, "args", _build_base_expr)),
-            scope=pgast.OptionsScope.TRANSACTION
-            if n["name"] in tx_only_vars or "is_local" in n and n["is_local"]
-            else pgast.OptionsScope.SESSION,
+            scope=(
+                pgast.OptionsScope.TRANSACTION
+                if n["name"] in tx_only_vars
+                or "is_local" in n
+                and n["is_local"]
+                else pgast.OptionsScope.SESSION
+            ),
```

Comment:
While here, could you add some parens? I don't like relying on the `or`/`and` precedence rule.
(I *think* and binds tighter (because it is more like multiplication?))